homeassistant/components/daikin/__init__.py

```diff
@@ -127,3 +130,71 @@ def device_info(self) -> DeviceInfo:
             name=info.get("name"),
             sw_version=info.get("ver", "").replace("_", "."),
         )
+
+
+async def async_migrate_unique_id(
+    hass: HomeAssistant, config_entry: ConfigEntry, api: DaikinApi
+) -> None:
+    """Migrate old entry."""
+    dev_reg = dr.async_get(hass)
+    old_unique_id = config_entry.unique_id
+    new_unique_id = api.device.mac
+    new_name = api.device.values.get("name")
```

Comment:
Should the new name be optional?

Reply:
I'll add a check for adding name when none

Reply:
The new name can still be optional with this call. If the key will always be present in the dict, please use `dict[key]` instead.

Reply:
The name was missing due to a bug so it should be updated in this case.

Reply:
I'm referring to the line here on 142.